### DISCUSSION TITLE
Handling app.state changes on db changes

### DISCUSSION THREAD

**Discussion by NixBiks** [litestar-org/litestar#3592] [Q&A] [answered]
Hi,

I'm new to `litestar` but it looks very neat. I have question about best practice for a use case I have.

I need to sync some data from my DB to `app.state` (low-latency requirements). 

I imagine adding `lifespan` context manager where I connect to the db, fetch initial data, set the initial `app.state` as well as setting up a websocket between the DB and my application - on each read I then need to update `app.state` but I'm not sure how to do this in a safe manner.

Any pointers for me?

**Comment by Alc-Alc** [comment, marked as answer]
Can you explain what you mean by "safe manner"? If you mean protection against data races, assignment to state via indexing (`state["something"] = "some value"`) and via attribute (`state.something = "somevalue"`) are protected with a lock making it thread safe. As for "async safe", nothing special has to be done because there is no context switching  / async operations inside the setters.

**Reply by Alc-Alc** [reply]
Relevant code

https://github.com/litestar-org/litestar/blob/de8f4a77ff1ef2095e2dc2657b77b9b866322d3f/litestar/datastructures/state.py#L254-L280
